libs/service/api_service.cpp

```diff
@@ -349,8 +354,13 @@ ebpf_verify_and_load_program(
                 goto Exit;
             }
 
-            for (uint32_t helper_id = 0; helper_id < helper_id_adddress.size(); helper_id++) {
-                if (ubpf_register(vm, helper_id, nullptr, reinterpret_cast<void*>(helper_id_adddress[helper_id])) < 0) {
+            for (uint32_t helper_id = 0; helper_id < helper_id_adddress_and_flags.size(); helper_id++) {
+                if (ubpf_register(
+                        vm,
+                        helper_id,
+                        nullptr,
+                        reinterpret_cast<void*>(std::get<0>(helper_id_adddress_and_flags[helper_id])),
+                        std::get<1>(helper_id_adddress_and_flags[helper_id])) < 0) {
```

Comment:
Since only the tail call is supposed to have this UNWIND_ON_SUCCESS requirement, and its a general function (implemented in EC itself), why not make that service and program modules aware of the speciality of this helper function and let it hardcode the flag in call to ubpf_register if the helper-id == tail-call? This way you wouldnt have to change the rest of the modules and save some code.